python_modules/dagster/dagster/_utils/caching_instance_queryer.py

```diff
@@ -618,64 +662,59 @@ def get_root_unreconciled_ancestors(
         elif not self.asset_partition_has_materialization_or_observation(asset_partition):
             return {asset_partition.asset_key}
 
-        time_or_dynamic_partitioned = isinstance(
-            self.asset_graph.get_partitions_def(asset_partition.asset_key),
-            (TimeWindowPartitionsDefinition, DynamicPartitionsDefinition),
+        parent_asset_partitions = self.asset_graph.get_parents_partitions(
+            dynamic_partitions_store=self,
+            current_time=self._evaluation_time,
+            asset_key=asset_partition.asset_key,
+            partition_key=asset_partition.partition_key,
+        ).parent_partitions
+
+        updated_or_missing_parent_asset_partitions = (
+            self._get_updated_or_missing_parent_asset_partitions(
+                asset_partition, parent_asset_partitions
+            )
         )
 
-        parent_asset_partitions_by_key: Dict[AssetKey, List[AssetKeyPartitionKey]] = defaultdict(
-            list
+        root_unreconciled_ancestors = (
+            {asset_partition.asset_key} if updated_or_missing_parent_asset_partitions else set()
         )
 
-        for parent in self.asset_graph.get_parents_partitions(
+        # recurse over parents
+        for parent in set(parent_asset_partitions) - updated_or_missing_parent_asset_partitions:
+            root_unreconciled_ancestors.update(
+                self.get_root_unreconciled_ancestors(asset_partition=parent)
+            )
+        return root_unreconciled_ancestors
+
+    def get_missing_or_updated_or_going_to_update_parents(
```

Comment:
I think this is a bit of a confusing formulation ("going to update parents" only makes sense if you understand the implementation details of the auto materialize logic, and is not actually accurate in cases where some of the parents don't have auto materialize policies.

I think it'd be better to just have this function focus on the missing / updated parents, then separately use the information we have available to us in the `conditions_for_candidate` area to determine which parents are going to be materialized. Then we can just union those sets (set of parents that have been updated more recently than X, as well as set of parents that are going to materialize on this tick).